pkgs/development/libraries/science/math/mkl/default.nix

```diff
@@ -51,18 +58,22 @@ stdenvNoCC.mkDerivation rec {
       cp license.txt $out/lib/
   '';
 
+  # fixDarwinDylibName fails for libmkl_cdft_core.dylib because the
+  # larger updated load commands do not fit. Use install_name_tool
+  # explicitly and ignore the error.
+  postFixup = stdenvNoCC.lib.optionalString stdenvNoCC.isDarwin ''
+      for f in $out/lib/*.dylib; do
+          install_name_tool -id $out/lib/`basename $f` $f || true
+      done
+      install_name_tool -change @rpath/libiomp5.dylib $out/lib/libiomp5.dylib $out/lib/libmkl_intel_thread.dylib
+      install_name_tool -change @rpath/libtbb.dylib $out/lib/libtbb.dylib $out/lib/libmkl_tbb_thread.dylib
+      install_name_tool -change @rpath/libtbbmalloc.dylib $out/lib/libtbbmalloc.dylib $out/lib/libtbbmalloc_proxy.dylib
```

Comment:
Are you sure we can modify binaries with the ISSL license agreement?

Reply:
The [ISSL](https://software.intel.com/en-us/license/intel-simplified-software-license) allows to "redistribute the software without modification".  I doubt that changing the install_name of the libraries can be considered as a modification of the software. Since we're not installing the libraries in a standard location, we can't redistribute them if we don't change the install name accordingly.

Reply:
If that would be a problem we would be pretty much screwed, because we use patchelf quite heavily for proprietary software.

Reply:
I am not a lawyer, but I also would consider the ELF header as part of the actual software.

Reply:
There is a precedent for build recipes that perform modification of mkl: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=intel-parallel-studio-xe#n523